code/game/machinery/embedded_controller/airlock_program.dm

```diff
@@ -89,8 +89,7 @@
 	if(!receive_tag) return
 
 	if(receive_tag==tag_chamber_sensor)
-		if(signal.data["pressure"])
-			memory["chamber_sensor_pressure"] = text2num(signal.data["pressure"])
+		memory["chamber_sensor_pressure"] = text2num(signal.data["pressure"])
```

Comment:
This I think is incorrect: if it sends a signal without this tag at all, this will reset the local pressure tally to `0` for no reason. Check `("pressure" in signal_data)`.

Reply:
All the other sensors for the exterior and interior do not have this check for a reason. 
The airlock program needs to know the current pressure even if it's null. Otherwise if the last value it got that was above 0 was over the target pressure, it will never complete it's cycle.

Reply:
Yes...which the check I am suggesting here does not interfere with. `"pressure" = null` is sent, and `null` is recorded if `"pressure"` is one of the keys transmitted. If `"pressure"` is not sent at all, you should not update to `null`.

Reply:
I disagree with that. 
If "pressure" isn't sent, that means you don't have a chamber sensor anymore, and it's required. So the pressure being changed to 0/null makes sense. Otherwise it'll stay stuck with the last reading, and that would also be wrong.

Reply:
No, if "pressure" isn't sent someone modified the radio components, which is completely valid. If the sensor is destroyed, then probably being stuck is not unreasonable. In any case, it's up to the airlock to handle destruction of the sensor or whatever gracefully. It can e.g. periodically send an update request, or abort, or just proceed, etc.

Reply:
I don't know where you're going with this, but you're really losing me here.

The sensor value being set to null is legal. As long as a message tagged with the sensor's tag is sending the message. And that even if it doesn't sends the "pressure" key. 
Try it yourself with a airlock setup with 3 sensors and 2 different atmospheres to change between. You'll see what I mean.

Reply:
I don't know how to explain myself better. If a signal does not contain the item `list("pressure" = null)` or `list("pressure" = number)`, the airlock controller should not update its pressure tally. The way to check for this is `("pressure" in item)`. 

Currently the check being used is `item["pressure"]`, which ignores the signals `list("pressure" = null)` and `list("pressure" = 0)` and as you point out this leads to bugs. My proposed check will not ignore those signals and hence should resolve the bug you describe. 

You ALSO want, for reasons that escape me, to have a signal of e.g. `list("foo" = 5)` make the controller update the internal pressure tally to `null` (this is what omitting the check I am suggesting does). I dislike this: it is not needed to solve the issue you have described; it leads to intuitive behavior. Note that arguing that this is somehow consistent with how the controller works currently is simply invalid: currently, the controller makes a check _even stronger than_ the one I propose, so it certainly does not update internal pressure tally when `list("foo" = 5)` is sent.

If this is still unclear, it would probably be more useful to discuss on discord or involve another maint.

Reply:
Look at this before dismissing this as "simply invalid" pls:
```
	else if(receive_tag==tag_exterior_sensor)
		memory["external_sensor_pressure"] = text2num(signal.data["pressure"])
	else if(receive_tag==tag_interior_sensor)
		memory["internal_sensor_pressure"] = text2num(signal.data["pressure"])
	else if(receive_tag==tag_exterior_door)
		memory["exterior_status"]["state"] = signal.data["door_status"]
		memory["exterior_status"]["lock"] = signal.data["lock_status"]
	else if(receive_tag==tag_interior_door)
		memory["interior_status"]["state"] = signal.data["door_status"]
		memory["interior_status"]["lock"] = signal.data["lock_status"]
```
All other sensors do it exactly like I changed the chamber sensor to for a reason.
Even the doors do it like that. 

That's a legal state of things that's already being handled by the controller. If a matching connected device doesn't send the tags that are expected, you don't have a reading to show. Don't just keep whatever is in memory as there is currently no up to date pressure readings.
It might have made sense if sensors sent anything else, or could just skip sending pressure updates as part of their normal operation. But that's not the case, and would obfuscate there being something interfering with the sensors messages.

I have no ideas what to tell you other than that.

Reply:
I'm fine with changing all these other places as well, but clearly it's outside the scope of this fix.

I regret not having refactored the airlock controller further (to use a bunch of radio components), which would have already forced it to function like I describe across the board.